app/src/main/java/com/example/mycleaninglog/MainActivity.kt

```diff
@@ -43,46 +44,69 @@ class MainActivity : ComponentActivity() {
                 viewModel.user = user
                 viewModel.listenToMyRooms()
             }
+
             val user by viewModel.users.observeAsState(initial = emptyList())
             val myRooms by viewModel.myRooms.observeAsState(initial = emptyList())
             val myCleaningTasks by viewModel.cleaningTasks.observeAsState(initial = emptyList())
 
             MyCleaningLogTheme {
                 // A surface container using the 'background' color from the theme
-                Surface(color = MaterialTheme.colors.background) {
-                    Column(
-                        modifier = Modifier
-                            .fillMaxSize()
-                            .background(MaterialTheme.colors.background)
-                            .padding(24.dp)
-                    ) {
-                        //calls the 1st level expandable cards. hardcoded because they should never change. These make up the main menu
+                if(user.isNullOrEmpty()) {
+                    signIn()
+                }
+                else{
```

Comment:
Careful here... prompting for login at the right time is a good idea.  However, since this is an if-else, that means, once the user has signed in, the else part will be skipped.  Don't you want to draw the screen after login?  Perhaps split this into two different functions.